crates/vm/src/arch/integration_api.rs

```diff
@@ -133,6 +136,14 @@ pub trait VmCoreChip<F, I: VmAdapterInterface<F>> {
         reads: I::Reads,
     ) -> Result<(AdapterRuntimeContext<F, I>, Self::Record)>;
 
+    fn execute_instruction2<Mem, Ctx>(
+        &mut self,
+        state: &mut VmExecutionState<Mem, Ctx>,
+        instruction: &Instruction<F>,
+    ) -> Result<()>
+    where
+        Mem: GuestMemory;
```

Comment:
just to save code, you could make the `todo!` the default impl

Reply:
yep, but having them separate makes it easy to search and resolve them one by one